docs/design/features/localization-options.md

```diff
@@ -0,0 +1,307 @@
+
+Localization for native components will be based on the user's locale.
+
+Standard methods for native localization use the user's locale. However, managed satellite assemblies also respect the thread's current culture. If the managed thread's current culture is not the same as the user's locale, this could result in mixed languages. Attempting to have the native components also follow the managed thread's culture would introduce issues and add significant complexity:
+  - Managed code may not be callable at the time a message needs to be displayed
+  - Some messages propagate up from the host (e.g. errors in `AssemblyDependencyResolver`) and host components do not have a simple way to access the managed thread's culture
+  - Not all platforms have a standard way to choose a specific language outside of process-wide configuration (e.g. `gettext` on Linux always uses the locale based on environment variables)
+
+### Translated assets
+
+All localizable resources need to be in the XLIFF file format (.xlf). New tooling will be required to convert from an untranslated base format to language-specific .xlf files and from the language-specific .xlf files to a format (.rc/.po/.strings, UTF-8/UTF-16) that will be compiled into resource libraries (or deployed directly) for each platform.
+
+The existing [xliff-tasks](https://github.com/dotnet/xliff-tasks) tooling supports conversion between managed resource files (.resx) and .xlf files and building satellite resource libraries. It is all MSBuild-based and has no concept of native resources or build processes. Extending it in a way that works naturally with CMake builds across platforms would be non-trivial.
+
+It is also an option to create tooling directly integrated in the dotnet/runtime repo itself. This would not be a generic and reusable component outside of the dotnet/runtime repo and its build system.
+
+### Platforms
+
+Each platform has its own standard way and file formats for handling localization. There are two main approaches that can be taken here:
+  1. Convert a base format to platform-specific localization formats and use platform-specific localization support
+      - Requires a tool/script to convert file formats at build time
+      - Uses platform support and standard localization methods that the .NET Core team will not need to maintain
+      - Different method for handling localized resources on each platform
+  2. Create a custom solution: storage format for string resources, tools for generation from the [translated assets (.xlf)](#translated-assets), and implementation for reading
+      - Requires design of completely custom support that will need to be maintained by the .NET Core team on all platforms
+      - Allows support for both string resources from file or memory
+      - All platforms could use the same method for handling localized resources
+
+For [single-file](#single-file) support, (1) would not be sufficient and a custom solution (2) would be required.
```

Comment:
It seems to me that single file with attached localized resources in a separate file would still be fine. Localization is kind of an extra that's not needed by all kinds of applications, so if we enabled the developers to choose whether to include localized resources or not.

Reply:
I definitely like this as another option. Especially since on Windows (where it's the most important consideration) the solution is simple via Win32 resources without a need for separate files and lot of custom code. So maybe on Linux/Mac we would require separate files.

Reply:
I think that's definitely a reasonable stance. Added a couple lines about single-file with separate localized resources.